src/main/java/com/google/api/codegen/util/nodejs/NodeJSNameFormatter.java

```diff
@@ -17,20 +17,29 @@
 import com.google.api.codegen.util.Name;
 import com.google.api.codegen.util.NameFormatter;
 import com.google.api.codegen.util.NamePath;
+import com.google.common.collect.ImmutableSet;
 
 /**
  * The NameFormatter for NodeJS.
  */
 public class NodeJSNameFormatter implements NameFormatter {
 
+  private String wrapIfKeywordOrBuiltIn(String name) {
+    if (RESERVED_IDENTIFIER_SET.contains(name)) {
+      return name + "_";
+    } else {
+      return name;
+    }
+  }
+
   @Override
   public String className(Name name) {
     return name.toUpperCamel();
   }
 
   @Override
   public String localVarName(Name name) {
-    return name.toLowerCamel();
+    return wrapIfKeywordOrBuiltIn(name.toLowerCamel());
   }
 
   @Override
```

Comment:
In NodeJS, the following code is syntactically correct.
```javascript
var foo = {};
foo.function = 'foo';
this.class = 'bar';
foo.prototype.return = function() {};
```

Therefore, field name, method name, etc wouldn't need that.
Static functions might need them.